store/v2/root/store.go

```diff
@@ -118,27 +118,46 @@ func (s *Store) SetInitialVersion(v uint64) error {
 	return s.stateCommitment.SetInitialVersion(v)
 }
 
-func (s *Store) StateLatest() (uint64, corestore.ReaderMap, error) {
-	v, err := s.GetLatestVersion()
+func (s *Store) getVersionedReader(version uint64) (store.VersionedReader, error) {
+	isExist, err := s.stateStorage.VersionExists(version)
 	if err != nil {
-		return 0, nil, err
+		return nil, err
+	}
+	if isExist {
+		return s.stateStorage, nil
 	}
 
-	return v, NewReaderMap(v, s), nil
+	if vReader, ok := s.stateCommitment.(store.VersionedReader); ok {
```

Comment:
we dont want to force commitment structures to implement iteration. This is here because of iavl mainly. If iavl didnt have iteration then this wouldnt be needed because iteration wouldnt be used in the state machine like today